sentinel-adapter/sentinel-apache-dubbo-adapter/src/main/java/com/alibaba/csp/sentinel/adapter/dubbo/SentinelDubboProviderFilter.java

```diff
@@ -61,6 +61,7 @@ String getInterfaceName(Invoker invoker, String prefix) {
     @Override
     public Result invoke(Invoker<?> invoker, Invocation invocation) throws RpcException {
         // Get origin caller.
+        InvokeMode invokeMode = RpcUtils.getInvokeMode(invoker.getUrl(), invocation);
```

Comment:
Thank you for pointing out the problem of the provider in the asynchronous scenario!

I test it and find two problems to be discussed.

If the interface method using `CompletableFuture<T>` as return type, `RpcUtils.getInvokeMode` will return `InvokeMode.FUTURE`, and `syncInvoke` will be called. In the scenario where the provider does asynchronous execution, should we use `invokemode. Future` to distinguish between sync  and async execution？

Another case is using `AsyncContext`, how can we judge whether it's a async execution in provider

Refer to [https://dubbo.apache.org/zh/docs/v2.7/user/examples/async-execute-on-provider/](https://dubbo.apache.org/zh/docs/v2.7/user/examples/async-execute-on-provider/)